### PR TITLE
feat(JSUI-2701): Update commerce fields and tests

### PR DESCRIPTION
**Pull request by louis-bompart**
Following the changes in the commerce rest parameter done in the backend with https://coveord.atlassian.net/browse/SEARCHAPI-4036, the `ICommerceRequest` interface needs to be updated to keep-up with the new specs.

All fields are considered optional because they can be also set in the search token (e.g: a catalogId is mandatory for any of the other field to be there, but it can be set in the search token, so it can be omitted in the query).

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

**Comment by louis-bompart** [comment]
At this moment, the changes are only available in dev.

File: src/rest/Query.ts [hunk 1]
```diff
@@ -34,7 +34,17 @@ export interface ICommerceRequest {
    *
    * **Example:** `46bc4275-e613-4dd5-b1ea-3e5aca1bcd9d`
    */
-  catalogId: string;
+  catalogId?: string;
+  /**
+   * A mandatory query expression applied if and only if a catalog affect the search query.
```
Comment (fbeaudoincoveo) [inline src/rest/Query.ts]: Suggestion:

> A mandatory query expression to apply if the commerce request affects the query.

File: src/rest/Query.ts [hunk 2]
```diff
@@ -34,7 +34,17 @@ export interface ICommerceRequest {
...
+   * **Example:** `@storeid==1001`
+   */
+  filter?: string;
+  /**
+   * How the catalog affects the results returned.
```
Comment (fbeaudoincoveo) [inline src/rest/Query.ts]: Suggestion:

> The way the commerce request should affect query results.

Comment (louis-bompart) [reply src/rest/Query.ts]: Done!

**Pull request merged**